packages/select/src/ResourceSelect.js

```diff
@@ -81,6 +81,18 @@ const ResourceSelect = ({
         },
       };
 
+      if (typeof data.variables === 'function') {
```

Comment:
We should be checking if `rest.parameters` is a function so that we know if we need to spread the params onto the  variables or call the function to spread.